src/aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -643,7 +643,6 @@ def _perform_dry_run(self):
         with LocalTransport() as transport:
             with SubmitTestFolder() as folder:
                 calc_info = self.presubmit(folder)
-                transport.chdir(folder.abspath)
```

Comment:
yes, not needed. 
In the new implementation, instead of changing directory here, `upload_calculation` will take it from `folder.abspath` (in the case of `dry_run`)